pool/store/mariadb_store.py

```
@@ -0,0 +1,250 @@
+import os
+import yaml
+from pathlib import Path
+from typing import Optional, Set, List, Tuple, Dict
+
+import asyncio
+import aiomysql
+import pymysql
+from blspy import G1Element
+from chia.pools.pool_wallet_info import PoolState
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.coin_solution import CoinSolution
+from chia.util.ints import uint64
+
+from .abstract import AbstractPoolStore
+from ..record import FarmerRecord
+from ..util import RequestMetadata
+
+pymysql.converters.encoders[uint64] = pymysql.converters.escape_int
+pymysql.converters.conversions = pymysql.converters.encoders.copy()
+pymysql.converters.conversions.update(pymysql.converters.decoders)
+
+# We load our configurations from here
+with open(os.getcwd() + "/config.yaml") as f:
+    pool_config: Dict = yaml.safe_load(f)
+#load db host from config
+db_host = pool_config["db_host"]
+#load db port from config
+db_port = pool_config["db_port"]
+#load db user from config
+db_user = pool_config["db_user"]
+#load db password from config
+db_password = pool_config["db_password"]
+#load db name from config
+db_name = pool_config["db_name"]
+
+class MariadbPoolStore(AbstractPoolStore):
+    """
+    Pool store based on MariaDB.
+    """
+    async def connect(self):
+        try:
+            self.pool = await aiomysql.create_pool(
+            minsize=1, 
+            maxsize=12,
+            host=db_host,
+            port=db_port,
+            user=db_user,
+            password=db_password,
+            db=db_name,
+            )
+        except aiomysql.Error.InterfaceError as e:
+                self.log.error("Error In Database Config. Check your config file! %s", e)
```

You should raise an exception here

---

Good point!

---

Added